code/modules/food_and_drink/pizza.dm

```diff
@@ -0,0 +1,593 @@
+	proc/add_topping(var/obj/item/topping, var/mob/user, var/list/params = null)
+		if(src.place_on(topping, user, params))
+			boutput(user, SPAN_NOTICE("You add \the [topping] to \the [src]."))
+			src.topping_space_left -= topping.w_class
+			topping.set_loc(src)
+			topping.transform = matrix(matrix(src.transform, src.topping_scale, src.topping_scale, MATRIX_SCALE), rand(-180,180), MATRIX_ROTATE)
+			topping.appearance_flags |= RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			topping.vis_flags |= VIS_INHERIT_PLANE | VIS_INHERIT_LAYER
+			topping.event_handler_flags |= NO_MOUSEDROP_QOL
+			src.vis_contents += topping
+			RegisterSignal(topping, COMSIG_ATOM_MOUSEDROP, PROC_REF(indirect_pickup))
+			RegisterSignal(topping, COMSIG_ATTACKHAND, PROC_REF(remove_topping))
+			src.UpdateIcon()
+		else
+			boutput(user, SPAN_ALERT("You can't do that, [topping] is attached to you!"))
+
+	/// Removes a topping from the pizza.
+	proc/remove_topping(var/obj/item/topping)
+		MOVE_OUT_TO_TURF_SAFE(topping, src)
+		src.vis_contents -= topping
+		src.topping_space_left += topping.w_class
+		topping.transform = initial(topping.transform)
+		topping.appearance_flags = initial(topping.appearance_flags)
+		topping.vis_flags = initial(topping.vis_flags)
+		topping.event_handler_flags = initial(topping.event_handler_flags)
+		UnregisterSignal(topping, COMSIG_ATOM_MOUSEDROP)
+		UnregisterSignal(topping, COMSIG_ATTACKHAND)
+
+		src.UpdateIcon()
+
+	attack_self(var/mob/user as mob)
+		if(user.traitHolder.hasTrait("training_chef") || prob(60))
+			user.visible_message(SPAN_NOTICE("[user] [pick("expertly", "acrobatically", "deftly", "masterfully")] spins \the [src]."), group = "pizzaspin")
+			src.fancy_spins++
+			if(src.fancy_spins == 10)
+				boutput(user,SPAN_ALERT("\The [src] is spun to perfection."))
+				if (prob(50))
+					JOB_XP(user, "Chef", 1)
+		else
+			user.visible_message(SPAN_ALERT("[user] [pick("clumsily", "catastrophically", "shamefully")] fumbles \the [src]."), group = "pizzaspin")
+			src.fancy_spins = 0
+			user.u_equip(src)
+			src.set_loc(user.loc)
+
+	attack(mob/target, mob/user, def_zone, is_special = FALSE, params = null)
+		if (user == target)
+			boutput(user, SPAN_ALERT("You need to at least bake it, you greedy beast!"))
+			user.visible_message("<b>[user]</b> stares at [src] in a confused manner.")
+			return
+		else
+			user.visible_message(SPAN_ALERT("<b>[user]</b> futilely attempts to shove [src] into [target]'s mouth!"))
+			return
+
+	Eat(mob/M as mob, mob/user, by_matter_eater = TRUE)
+		if(!by_matter_eater)
+			return 0
+		var/obj/item/reagent_containers/food/snacks/pizza/unbaked_pizza = src.bake_pizza(TRUE)
+		return unbaked_pizza.Eat(M, user, TRUE)
+
+	temperature_expose(datum/gas_mixture/air, temperature, volume)
+		if (temperature >= T0C+3200) // syndicate zippos and raging plasmafires, for laughs
+			var/obj/item/reagent_containers/food/snacks/pizza/baked_pizza = src.bake_pizza()
+			baked_pizza.quality = 5
```

Comment:
could use some an audio/chatbox cue when this happens

Reply:
Sure thing!